core/include/scipp/core/view_index.h

```diff
@@ -104,15 +105,25 @@ class SCIPP_CORE_EXPORT ViewIndex {
   // as the performance is identical to C-style arrays, as long as range based
   // loops are used.
 
-  scipp::index m_index{0};
+  /// Index into memory.
+  scipp::index m_memory_index{0};
+  /// Steps to advance one element.
   std::array<scipp::index, NDIM_MAX> m_delta = {};
+  /// Multi-dimensional index in iteration dimensions.
   std::array<scipp::index, NDIM_MAX> m_coord = {};
-  std::array<scipp::index, NDIM_MAX> m_extent = {};
-  scipp::index m_fullIndex;
-  int32_t m_dims;
-  int32_t m_subdims = 0;
-  std::array<int32_t, NDIM_MAX> m_offsets;
-  std::array<scipp::index, NDIM_MAX> m_factors;
+  /// Shape in iteration dimensions.
+  std::array<scipp::index, NDIM_MAX> m_shape = {};
+  /// Strides in memory.
+  Strides m_strides;
```

Comment:
Do we need `m_strides` AND `m_delta`?

Reply:
Strictly speaking, no. But this should improve performance. Each one is beneficial for only one of sequential and random lookup.
I will benchmark and see if this really matters.